skimage/util/process.py

```diff
@@ -0,0 +1,68 @@
+from multiprocessing import cpu_count
+
+import dask.array as da
+
+__all__ = ['process_chunks']
+
+
+def _get_chunks(shape, ncpu):
```

Comment:
Storage formats like `h5py` have a bit of an easier time with chunk sizes, they can rely on them being uniform.  When you start actually computing on your arrays your chunk sizes can change and these changes can be heterogeneous.

Chunks is defined here http://dask.pydata.org/en/latest/array-design.html and generally means the sizes of each block along each dimension.  These can vary along each axis but must be Cartesian within the array.